compiler/typecheck/TcGenGenerics.hs

```diff
@@ -87,6 +87,7 @@ get_gen1_constrained_tys :: TyVar -> Type -> [Type]
 get_gen1_constrained_tys argVar
   = argTyFold argVar $ ArgTyAlg { ata_rec0 = const []
                                 , ata_par1 = [], ata_rec1 = const []
+                                , ata_parAp1 = id
```

Comment:
This should be `const []`, nothing under the parameter will ever be able to be reached, just like `ata_rec1`.